geo/src/algorithm/coordinate_position.rs

```diff
@@ -371,21 +366,30 @@ where
     // See: https://en.wikipedia.org/wiki/Point_in_polygon#Winding_number_algorithm
     let mut winding_number = 0;
     for line in linestring.lines() {
+        // Edge Crossing Rules:
+        //   1. an upward edge includes its starting endpoint, and excludes its final endpoint;
+        //   2. a downward edge excludes its starting endpoint, and includes its final endpoint;
+        //   3. horizontal edges are excluded
+        //   4. the edge-ray intersection point must be strictly right of the point P.
         if line.start.y <= coord.y {
-            let o = T::Ker::orient2d(line.start, line.end, coord);
-            if o == Orientation::Collinear && point_in_rect(coord, line.start, line.end) {
-                return CoordPos::OnBoundary;
-            }
-            if line.end.y > coord.y && o == Orientation::CounterClockwise {
-                winding_number += 1;
-            }
+            if line.end.y >= coord.y {
```

Comment:
I'm trying to understand why these changes make things faster. 

With this new check `line.end.y >= coord.y`, we ensure we only call orient2d if a horizontal ray passing through `coord` would also pass through `line`. Checking `line.end.y >= coord.y` is much cheaper than orient2d, so we gain speed there.

But then, shouldn't we have a similar check (`if line.start.y >= coord.y {`) for the downward edge case below?

I confess that I tried it, but saw somewhere between small and no gains. I don't understand why though.

Reply:
Its basically this, that for all edges with starting y coord below the ray, the orientation was calculated before checking for the line ending above the line.

> But then, shouldn't we have a similar check (`if line.start.y >= coord.y {`) for the downward edge case below?

This is implicitly given by the test in the if clause `line.start.y <= coord.y` which gives us `line.start.y > coord.y` in the else clause what matches with the edgecrossing rule 2. a downward edge excludes its starting endpoint.

Reply:
oh duh, `line.start.y >= coord.y` is always true at that point. Thank you. =)